core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java

```diff
@@ -184,8 +203,19 @@ private static boolean isRecorder(AnnotatedElement element) {
         final DefaultValuesConfigurationSource ds2 = new DefaultValuesConfigurationSource(
                 reader.getBuildTimeRunTimePatternMap());
         final PropertiesConfigSource pcs = new PropertiesConfigSource(buildSystemProps, "Build system");
-
-        builder.withSources(ds1, ds2, pcs);
+        if (platformProperties.isEmpty()) {
+            builder.withSources(ds1, ds2, pcs);
+        } else {
+            final KeyMap<String> props = new KeyMap<>(platformProperties.size());
+            for (Map.Entry<String, String> prop : platformProperties.entrySet()) {
+                props.findOrAdd(prop.getKey().split("\\.")).putRootValue(prop.getValue());
```

Comment:
```suggestion
                props.findOrAdd(new NameIterator(prop.getKey())).putRootValue(prop.getValue());
```
You'll have to add an import for `NameIterator` too.  But it occurred to me that `NameIterator` already handles interior `"` and escaping for you, and is more efficient than split anyway.

Reply:
Thanks, @dmlloyd! Applied.